src/main/java/edu/hm/hafner/analysis/parser/SimulinkCheckParser.java

```diff
@@ -0,0 +1,131 @@
+
+import edu.hm.hafner.analysis.IssueBuilder;
+import edu.hm.hafner.analysis.ParsingException;
+import edu.hm.hafner.analysis.ReaderFactory;
+import edu.hm.hafner.analysis.Report;
+import edu.hm.hafner.analysis.Severity;
+
+/**
+ * A parser for Simulink Check tool by Mathworks results.
+ * Used for .html report files.
+ *
+ * @author Eva Habeeb
+ */
+public class SimulinkCheckParser extends IssuePropertiesParser {
+
+    private static final long serialVersionUID = -8099258658775128275L;
+
+    @Override
+    @SuppressWarnings({"PMD.NcssCount", "PMD.CyclomaticComplexity", "PMD.CognitiveComplexity", "PMD.NPathComplexity" })
+    public Report parse(final ReaderFactory readerFactory) throws ParsingException {
+        String name = readerFactory.getFileName();
+        File file = new File(name);
+
+        try (IssueBuilder issueBuilder = new IssueBuilder()) {
+            //Parse html document using Jsoup
+            Document document = Jsoup.parse(file, "utf-8");
+            Elements systemElement = document.select("div.ReportContent");
+            String system = systemElement.first().id();
+            Report report = new Report();
+
+            //Select Warning items from the whole html document
+            parseIssue(report, document, issueBuilder, system, "div.WarningCheck");
+            //Select Failed items from the whole html document
+            parseIssue(report, document, issueBuilder, system, "div.FailedCheck");
+            //Select Not Run items from the whole html document
+            parseIssue(report, document, issueBuilder, system, "div.NotRunCheck");
+            //Select Incomplete items from the whole html document
+            parseIssue(report, document, issueBuilder, system, "div.IncompleteCheck");
+
+            return report;
+        }
+        catch (IOException e) {
+            throw new ParsingException(e);
+        }
+    }
+
+    /**
+     * Handles parsing.
+     * @param report the report object
+     * @param document parsed html document
+     * @param issueBuilder issueBuilder object
+     * @param system name of the system
+     * @param check category of the check
+     *
+     */
+    public void parseIssue(final Report report, final Document document, final IssueBuilder issueBuilder, final String system, final String check) {
+        Elements heading;
+        String issueTxt;
+        String headingElement;
+        String[] headingSplit;
+        int strLen;
+        String module;
+
+        Elements issueElements = document.select(check);
+        for (Element w : issueElements) {
+            heading = w.select("span.CheckHeading");
+            issueTxt = heading.text();
+            setSeverity(check, issueBuilder);
+
+            headingElement = heading.first().id();
+            headingSplit = headingElement.split("\\.");
+            strLen = headingSplit.length;
+
+            if (issueTxt.contains("SW")) {
+                String[] fileName = issueTxt.split(":", 2);
```

Comment:
What happens, if not present? Then the index is not accessible

Reply:
This case is added for a specific scenario where the issue text will for sure have the value SW0 and ':', otherwise it will be handled by the other else statements. I have updated the condition to "if (issueTxt.contains("SW0"))" to make sure it is the correct case.

Reply:
But when they decide to use a - in a future version of the tool then your code breaks with an `IndexOutOfBoundsException`.